python/cudf/cudf/_lib/copying.pyx

```diff
@@ -202,17 +218,15 @@ def copy_range(out_col, in_col, int out_begin, int out_end,
     if out_begin > out_end:
         return out_col
 
-    if out_col.null_count == 0 and in_col.has_null_mask:
+    if out_col.null_count == 0 and in_col.mask:
         mask = utils.make_mask(len(out_col))
         cudautils.fill_value(mask, 0xff)
-        out_col._mask = Buffer(mask)
-        out_col._null_count = 0
```

Comment:
How is null count tracked now? This makes me think it's going to be computed via the mask.

Reply:
Done. We now use a `cached_property` to compute the `null_count` which is invalidated whenever we set `.mask` or get a mutable view of the column